### ISSUE TITLE
hsluv roundtrip not working

### ISSUE THREAD

**Issue by varon** [williammalo/hsluv-glsl#1]
I'm not able to get a round-trip working from `rgb -> hsluv -> rgb`. is this is a usage error, perhaps?

I receive a greyscale values, as though the saturation component is mistakingly zeroed.

For a test reproduction, you can view the following shader toy, which has the same behaviour: https://www.shadertoy.com/view/XlKXz1

Other information:
   Desktop OpenGL
    Context: 4.4.0 NVIDIA 369.09
    GLSL Version: 4.40 NVIDIA via Cg compiler
    Renderer: GeForce GTX 670/PCIe/SSE2

**Comment by varon** [comment]
Seems like there's a multiplication sign issue.
In the sample code, swapping the round trip to this solves the issue.
```
    vec4 roundTrip = hsluvToRgb(hsluv.x, 10000.0*hsluv.y, hsluv.z, hsluv.w);

**Comment by varon** [comment]
Swapping
```
vec3 lchToHpluv(vec3 tuple) {
    tuple.g /= hsluv_maxSafeChromaForL(tuple.r) * 100.0;
    return tuple.bgr;
}
```
to
```
vec3 lchToHpluv(vec3 tuple) {
    tuple.g /= hsluv_maxSafeChromaForL(tuple.r) / 100.0;
    return tuple.bgr;
}
```
solves the issue for me.

edit: (for hpluv)

**Comment by varon** [comment]
swapping
```
vec3 hsluvToLch(vec3 tuple) {
    tuple.g *= hsluv_maxChromaForLH(tuple.b, tuple.r) / 100.0;
    return tuple.bgr;
}
```
to
```
vec3 hsluvToLch(vec3 tuple) {
    tuple.g *= hsluv_maxChromaForLH(tuple.b, tuple.r) * 100.0;
    return tuple.bgr;
}
```

solves the hsluv

**Comment by williammalo** [comment]
Ah, thanks! I will look into it

**Comment by heisters** [comment]
This commit breaks the code for me. Or am I calling it incorrectly?

This test tries to output an HSLuv rainbow, with and without a roundtrip through HSLuv. 

https://www.shadertoy.com/view/XdlfR4

The shadertoy shows two failing tests in positions 1 and 3, and two passing tests in positions 2 and 4. The passing tests look like smooth gradients through the HSLuv hues.

<img width="643" alt="screen shot 2017-05-17 at 4 25 17 pm" src="https://cloud.githubusercontent.com/assets/72790/26180284/7bf507dc-3b1d-11e7-8fda-a3d100df772e.png">

Swapping the scaling from multiplication to division in hsluvToLch and lchToHsluv fix the issue for me (see `hsluvToLchFIXED` and `lchToHsluvFIXED` in the example).

Generally, are inputs supposed to be in HSLuv ranges (0 - 360 for H, 0 - 100 for S and L), or in normalized coordinates (0 - 1)?

**Comment by williammalo** [comment]
I will look into it

**Comment by williammalo** [comment]
@heisters fixed in v4.2
